catalog/app/containers/Bucket/viewModes.ts

```diff
@@ -36,6 +36,7 @@ export function viewModeToSelectOption(m: ViewMode | null): SelectOption | null
 export function useViewModes(
   path: string,
   modeInput: string | null | undefined,
+  // XXX: consider using a plain boolean here since the contents of this object are unused
```

Comment:
It's a deliberate choice. Reusing a small number of data structures reduces cognitive load. Also, it reduces footprint because it doesn't copy data, just passes the link to the object.